modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/SynonymTokenFilterFactory.java

```diff
@@ -33,6 +33,19 @@
 import java.util.function.Function;
 
 public class SynonymTokenFilterFactory extends AbstractTokenFilterFactory {
+    private enum SynonymsSource {
```

Comment:
I've found [this enum pattern](https://github.com/elastic/elasticsearch/blob/9ba5651e7478216f17d81b63385ddf6db2331f7f/server/src/main/java/org/elasticsearch/Build.java#L152) more prominent in the codebase - using a string ctor and a displayName() method for retrieving it

Reply:
Much better way to do it 🤦